### PR TITLE
Hsearch 3456 Expose meta-information about the query execution (execution time, timeout, ...)

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-3456

### CODE REVIEW

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/ElasticsearchSearchResult.java [hunk 1]
```diff
@@ -9,4 +9,9 @@
 import org.hibernate.search.engine.search.query.SearchResult;
 
 public interface ElasticsearchSearchResult<H> extends SearchResult<H> {
+
+	Integer getTook();
+
+	Boolean getTimedOut();
```
Comment (yrodiere) [inline backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/ElasticsearchSearchResult.java]: Maybe `isTimedOut` instead? And return a primitive boolean?

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/ElasticsearchSearchResult.java [hunk 2]
```diff
@@ -9,4 +9,9 @@
 import org.hibernate.search.engine.search.query.SearchResult;
 
 public interface ElasticsearchSearchResult<H> extends SearchResult<H> {
+
+	Integer getTook();
```
Comment (yrodiere) [inline backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/ElasticsearchSearchResult.java]: Maybe return a Duration instead?

Comment (yrodiere) [reply backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/ElasticsearchSearchResult.java]: Also some javadoc would be nice.

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/ElasticsearchSearchResult.java (hunk 1 above)
Comment (yrodiere) [reply backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/ElasticsearchSearchResult.java]: Also some javadoc would be nice.

File: integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/ElasticsearchExtensionIT.java [hunk 3]
```diff
@@ -171,6 +172,21 @@ public void query() {
...
+
+		SearchQuery<DocumentReference> genericQuery = scope.query()
+				.predicate( f -> f.matchAll() )
+				.toQuery();
+
+		ElasticsearchSearchQuery<DocumentReference> esQuery = genericQuery.extension( ElasticsearchExtension.get() );
+		ElasticsearchSearchResult<DocumentReference> result = esQuery.fetchAll();
+
+		assertNotNull( result.getTook() );
+		assertNotNull( result.getTimedOut() );
```
Comment (yrodiere) [inline integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/ElasticsearchExtensionIT.java]: You could even assert it's false, in this case :)

File: documentation/src/main/asciidoc/search-dsl.asciidoc [hunk 4]
```diff
@@ -1575,6 +1575,20 @@ include::{sourcedir}/org/hibernate/search/documentation/search/projection/Projec
 ----
 ====
 
+==== Elasticsearch: `took` and `explanation`
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/search-dsl.asciidoc]: This has nothing to do with projections, so it shouldn't be in the section of the documation relative to projections.
Maybe put this in the section about the query DSL, after the section `search-dsl-query-debugging-scores`?

File: documentation/src/test/java/org/hibernate/search/documentation/search/query/QueryDslIT.java [hunk 5]
```diff
@@ -306,6 +307,31 @@ public void explain_elasticsearch() {
 		} );
 	}
 
+	@Test
+	public void tookAndTimedOut_elasticsearch() {
```
Comment (yrodiere) [inline documentation/src/test/java/org/hibernate/search/documentation/search/query/QueryDslIT.java]: See, you even put the example in the integration test about the query DSL (not the projection DSL) :)

File: integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/ElasticsearchExtensionIT.java (hunk 3 above)
Comment (fax4ever) [reply integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/ElasticsearchExtensionIT.java]: Yes, I thought about that, but I was not sure about to do that.

**Comment by fax4ever** [comment]
Changes addressed. I don't know if there's anything else to do. Thanks

**Comment by fax4ever** [comment]
Merged thanks!

**Pull request merged**